### PR TITLE
fixed command line api and updated readme

### PR DESCRIPTION
**Pull request by jo-pol**
the new optional boolean argument caused problems and error reporting gave no clues
- changed tailArgs to options with custom converters for DateTime and not existing files
- added test to check that readme was properly maintained
- made settings backward compatible for EasyIngestFlow with a default value for the new argument
- changed home variable to system property app.home (like easy-update-solr-index)

### CODE REVIEW

**Comment by jo-pol** [comment]
@GeorgiKhomeriki @janvanmansum @PaulBoon @ekoi @vesaakerman
please have a look

File: src/main/scala/nl/knaw/dans/easy/stage/Conf.scala [hunk 1]
```diff
@@ -3,57 +3,45 @@ package nl.knaw.dans.easy.stage
 import java.io.File
 
 import org.joda.time.DateTime
-import org.rogach.scallop.ScallopConf
+import org.rogach.scallop.{singleArgConverter, ValueConverter, ScallopConf}
 
-class Conf(args: Seq[String]) extends ScallopConf(args) {
-  printedName = "easy-state-dataset"
+class Conf(args: Seq[String] = "-b. -s. -t2015 -uu -dd".split(" ")) extends ScallopConf(args) {
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:8]: Bit of a convoluted and implicit way to set defaults for some of the parameters. Also, I am not sure what -uu and -dd actually will do.

File: src/main/scala/nl/knaw/dans/easy/stage/Conf.scala [hunk 2]
```diff
@@ -3,57 +3,45 @@ package nl.knaw.dans.easy.stage
...
-   * Scallop fails with a vague error if a type File argument refers to a file that does
-   * not (yet) exist. As the SDO-set might not yet exist we work around this by making
-   * this a type String argument
-   */
-  val sdoSet = trailArg[String](
-    name = "staged-digital-object-set",
+  val sdoSet = opt[File](
+    name = "staged-digital-object-set", short = 's',
     descr = "The resulting Staged Digital Object directory (will be created if it does not exist)",
+    required = true)(singleArgConverter[File](conv = new File(_)))
```
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:30]: @janvanmansum 
This converter allows a directory that does not exist. I did not make it implicit like the DateTime converter as the file for sdoSet should exist. It results in clearer error messages than the vague "could not parse trailArg"

Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:30]: The following might have been clearer:

```
val mayNotExist = singleArgConverter[File](conv = new File(_))
...
val sdoSet = opt[File](...)(mayNotExist)
```

Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/Conf.scala:30]: Sure, if that works, we'll do that.

**Pull request merged**